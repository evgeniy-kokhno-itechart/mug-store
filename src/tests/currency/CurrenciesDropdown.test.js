/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import CurrensiesDropdown from '../../app/currency/components/CurrenciesDropdown/CurrenciesDropdown';

configure({ adapter: new Adapter() });

describe('<CurrensiesDropdown />', () => {
  let CurrensiesDropdownWrapper;

  beforeEach(() => {
    const onCurrencyChange = jest.fn();
    CurrensiesDropdownWrapper = shallow(<CurrensiesDropdown onChange={onCurrencyChange} />);
  });

  it('renders spinner with default props provided', () => {
    expect(CurrensiesDropdownWrapper).toMatchSnapshot();
  });

  it('renders currencies dropdown with currencies provided', () => {
    CurrensiesDropdownWrapper.setProps({
      currencyState: {
        currentCurrency: { id: '1' },
        currencies: [{ id: '1', name: 'test' }],
        currenciesStatus: {
          isInProcess: false,
          hasFailed: false,
          error: '',
        },
      },
    });
    expect(CurrensiesDropdownWrapper).toMatchSnapshot();
  });

  it('renders badge within error with when currenciesStatus.hasFailed=true and errorMessage provided', () => {
    CurrensiesDropdownWrapper.setProps({
      currencyState: {
        currentCurrency: { id: '1' },
        currencies: [{ id: '1', name: 'test' }],
        currenciesStatus: {
          isInProcess: false,
          hasFailed: true,
          error: 'test error',
        },
      },
    });
    expect(CurrensiesDropdownWrapper).toMatchSnapshot();
  });
});
