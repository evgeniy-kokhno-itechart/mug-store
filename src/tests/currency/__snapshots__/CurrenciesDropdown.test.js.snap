// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrensiesDropdown /> renders badge within error with when currenciesStatus.hasFailed=true and errorMessage provided 1`] = `
<span
  className="badge badge-danger"
>
  test error
</span>
`;

exports[`<CurrensiesDropdown /> renders currencies dropdown with currencies provided 1`] = `
<DropdownInline
  defaultText=""
  error=""
  label="Currency:"
  labelClasses="d-inline"
  name="currencyDropdown"
  onChange={[Function]}
  options={
    Array [
      Object {
        "id": "1",
        "name": "test",
      },
    ]
  }
  selectClasses="text-white bg-dark border-0 mr-3"
  value="1"
  wrapperClasses="navbar-text d-inline"
/>
`;

exports[`<CurrensiesDropdown /> renders spinner with default props provided 1`] = `
<Spinner
  spinnerClasses="spinner-border-sm text-light"
  wrapperClasses="d-inline mr-3"
/>
`;
